src/sentry/integrations/utils/code_mapping.py

```diff
@@ -0,0 +1,184 @@
+                bucket_key = frame_filename.root if frame_filename.root else NO_TOP_DIR
+
+                if not buckets.get(bucket_key):
+                    buckets[bucket_key] = []
+                buckets[bucket_key].append(frame_filename)
+
+            except ValueError:
+                logger.exception(
+                    f"We cannot breakdown this stacktrace path: {stacktrace_frame_file_path}"
+                )
+                continue
+        return buckets
+
+    def generate_code_mappings(self, stacktraces: List[str]) -> List[CodeMapping]:
+        """Generate code mappings based on the initial trees object and the list of stack traces"""
+        # We need to make sure that calling this method with a new list of stack traces
+        # should always start with a clean slate
+        self.code_mappings = {}
+        buckets: Dict[str, Any] = self.stacktrace_buckets(stacktraces)
+
+        for stackframe_root, stackframes in buckets.items():
+            if not self.code_mappings.get(stackframe_root):
+                for frame_filename in stackframes:
+                    code_mapping = self._find_code_mapping(frame_filename)
+                    if code_mapping:
+                        # XXX: Reprocess feature
+                        # XXX: Reverse mapping may need to be on a per-repo basis
+                        # This helps excluding some source files when we have a matching code mapping
+                        self.reverse_mapping.append(code_mapping.source_path)
+                        self.code_mappings[stackframe_root] = code_mapping
+
+        return list(self.code_mappings.values())
+
+    def _find_code_mapping(self, frame_filename: FrameFilename) -> Union[CodeMapping, None]:
+        """Look for the file path through all the trees and generate code mappings for it"""
+        _code_mappings: List[CodeMapping] = []
+        # XXX: This will need optimization by changing the data structure of the trees
+        for repo_full_name, tree in self.trees.items():
+            _code_mappings.extend(
+                self._generate_code_mapping_from_tree(repo_full_name, frame_filename)
+            )
+
+        if len(_code_mappings) == 0:
+            logger.warning(f"No files matched for {frame_filename.full_path}")
+            return None
+        # This means that the file has been found in more than one repo
+        elif len(_code_mappings) > 1:
+            logger.warning(f"More than one file matched for {frame_filename.full_path}")
+            return None
+
+        return _code_mappings[0]
+
+    def _generate_code_mapping_from_tree(
+        self,
+        repo_full_name: str,
+        frame_filename: FrameFilename,
+    ) -> List[CodeMapping]:
+        matched_files = list(
+            filter(
+                lambda src_path: self._potential_match(src_path, frame_filename),
+                self.trees[repo_full_name],
+            )
+        )
```

Comment:
List comprehensions are generally considered more Pythonic and a bit more performant. ([interesting read](https://stackoverflow.com/questions/3013449/list-comprehension-vs-lambda-filter))

```suggestion
       matched_files=[self._potential_match(src_patch, frame_filename) for src_path in self.trees[repo_full_name]]
```

Reply:
The JS in me 😅

Reply:
I had to move back to my original approach since it was not working as expected. I will handle it later.